Product/Production/Adapters/General/CONNECTAdminGUI/src/main/java/gov/hhs/fha/nhinc/admingui/services/GatewayService.java

```diff
@@ -175,10 +186,20 @@ public boolean retrieveDocument(PatientSearchBean patientQuerySearch) {
 
         //Call the NwHIN service to retrieve the document
         DocumentRetrieveResults response = documentRetrieveService.retrieveDocuments(docRetrieve);
-        //set the retrieved document to the UI patient bean 
+        //set the retrieved document to the UI patient bean
         if (response.getDocument() != null) {
-            //patientQuerySearch.getSelectedCurrentDocument().setContentType(response.getContentType());
-            patientQuerySearch.getSelectedCurrentDocument().setDocumentContent(response.getDocument());
+            if (response.getContentType().contains("text/xml")) {
+                InputStream xsl = FacesContext.getCurrentInstance().getExternalContext().getResourceAsStream("/WEB-INF/CDA.xsl");
+                InputStream xml = new ByteArrayInputStream(response.getDocument());
+                byte[] convertXmlToHtml = null;
+                if (xsl != null) {
+                    convertXmlToHtml = convertXMLToHTML(xml, xsl);
+                    xsl.close();
+                }
+                patientQuerySearch.getSelectedCurrentDocument().setDocumentContent(convertXmlToHtml);
+            } else {
+                patientQuerySearch.getSelectedCurrentDocument().setDocumentContent(response.getDocument());
```

Comment:
This seems overly broad xsl for xml.  Can we refine this to just use on CDAs or even on specific organizations.  Is pdf and text handled by the else?